### PR TITLE
fix: break hard dependency on mutation options interface

### PR DESCRIPTION
**Pull request by StephenCoady**
Enable type safety for offlineMutation by conforming to Apollo mutate interface.

### CODE REVIEW

File: packages/offix-cache/src/cache/mutations.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export interface MutationHelperOptions extends MutationOptions {
  * Provides useful helpers for cache updates, optimistic responses, and context
  * @param options see `MutationHelperOptions`
  */
-export const createMutationOptions = (options: MutationHelperOptions): MutationOptions => {
+export const createMutationOptions = (options: MutationOptions & MutationHelperOptions): MutationOptions => {
```
Comment (wtrocki) [inline packages/offix-cache/src/cache/mutations.ts]: ah.. that is even nicer. This will decorate MutationHelperOptions and still work with <T> depedency.

**Comment by wtrocki** [comment]
@StephenCoady Can you also bump versions of the packages to see if this is going to compile with latest Apollo 2.6?

File: packages/offix-client/src/ApolloOfflineClient.ts [hunk 2]
```diff
@@ -28,6 +28,6 @@ export interface ApolloOfflineClient extends ApolloClient<NormalizedCacheObject>
    * Allows the client to perform an offline mutation
    * @param options the mutation helper options used to build the offline mutation
    */
-  offlineMutation<T>(options: MutationHelperOptions): Promise<FetchResult<T>>;
+  offlineMutation<T>(options: MutationOptions & MutationHelperOptions): Promise<FetchResult<T>>;
```
Comment (StephenCoady) [inline packages/offix-client/src/ApolloOfflineClient.ts]: @wtrocki we should probably reuse the merged type

File: packages/offix-client/src/OfflineClient.ts [hunk 3]
```diff
@@ -96,7 +98,7 @@ export class OfflineClient implements ListenerProvider {
    * this offline friendly function to handle the optimistic UI and cache updates.
    * @param options the MutationHelperOptions to create the mutation
    */
-  public offlineMutation<T>(options: MutationHelperOptions): Promise<FetchResult<T>> {
+  public offlineMutation<T>(options: MutationOptions & MutationHelperOptions): Promise<FetchResult<T>> {
```
Comment (StephenCoady) [inline packages/offix-client/src/OfflineClient.ts]: @wtrocki and here :P

**Pull request merged**